torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -1306,6 +1306,30 @@ at::Tensor XLANativeFunctions::expand(const at::Tensor& self,
       bridge::GetXlaTensor(self), torch::lazy::ToVector<int64_t>(size)));
 }
 
+at::Tensor XLANativeFunctions::expand_symint(const at::Tensor& self,
+                                             c10::SymIntArrayRef size,
```

Comment:
We aim to have all of the complicated logic in the Node instead of `aten_xla_type.cpp` since we want to codegen `aten_xla_type.cpp`.

we need to have Node with different size SymInt to has different hash, we can either
1. manually expand `Symint` to upper_bound + is_dynamic + Node
2. Implement a hash functions for `SymInt` that account the hash for above 3 componenets.


I think it is cleaner to have `aten_xla_type` to pass `c10::SymIntArrayRef` to `ExapndSymint` Node and handle all of the complicated logic in lowering. This way we can codegen this op later and handle all of the expand logic in place.

Reply:
I agree. I see three solutions here:

+ move the logic to `ExpandSymInt`. Moving the entire logic to this calss poses a bit of challenge because I need to pass the `const std::vector<torch::lazy::Value>& dimensions` to the constructor initialize the corresponding `XlaNode`. `dimensions` is the IRValue object from upstream Node. 

+ move the logic to `tensor_methods`. This is not the cleanest approach, but the easiest.

+ Something in between: produce the dimensions IR value in `tensor_methods` and unpack the rest (i.e. upper_bound, is_dynamic) in `ExpandSymInt`

I prefer the last choice. Wdyt @JackCaoG?

Reply:
The node is only useful in its' IR Value form. So, I decided to abstract the logic and make tensor_methods leaner. I will push the new change shortly.

Reply:
I addressed this as per our discussion offline.